### DISCUSSION TITLE
Table Rendering Bug

### DISCUSSION THREAD

**Discussion by natewiebe13** [gotenberg/gotenberg#393] [Q&A] [answered]
When a table thead is split across pages, the second page runs into some rendering issues. I've attached an example HTML file and the output I receive. See the top of page 2 of the PDF.

Here are the commands to reproduce:
```bash
docker run --rm -p 3000:3000 gotenberg/gotenberg:7

curl \
--request POST 'http://localhost:3000/forms/chromium/convert/html' \
--form 'files=@"/path/to/file/index.html"' \
--form 'paperWidth="8.5"' \
--form 'paperHeight="11"' \
--form 'marginTop="1"' \
--form 'marginBottom="1"' \
--form 'marginLeft="1"' \
--form 'marginRight="1"' \
-o output.pdf
```
Files:
[index.txt](https://github.com/gotenberg/gotenberg/files/7695127/index.txt)
[output.pdf](https://github.com/gotenberg/gotenberg/files/7695125/output.pdf)

**Note:** I had to rename `index.html` to `index.txt` in order to attach to the issue.

**UPDATE**

This now appears to be fixed.

**Comment by gulien** [comment]
Hello @natewiebe13 👋 

It’s not really an issue with Gotenberg but more how Chrome renders the HTML to PDF.  I think adding the following CSS could solve your problem:

 ```
thead { display: table-header-group }
tfoot { display: table-row-group }
tr { page-break-inside: avoid
```

**Comment by natewiebe13** [comment]
Unfortunately that CSS made no difference. I've attached updated input/output to this comment.

[index2.txt](https://github.com/gotenberg/gotenberg/files/7695766/index2.txt)
[output2.pdf](https://github.com/gotenberg/gotenberg/files/7695767/output2.pdf)

**Comment by gulien** [comment]
@natewiebe13 it seems to be a recurring issue when converting HTML tables to PDF. There are some solutions listed in the following links:

* https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2367
* https://www.api2pdf.com/solution-for-overlapping-content-when-generating-pdfs-with-table-headers-and-footers/
* https://blog.dev030.com/posts/repeating-table-headers-pdf-downloads-using-headless-chrome

As this is not really an issue with Gotenberg, I’m moving this to the discussion space ^^

Let me know if you find a solution to your problem 😊

**Comment by natewiebe13** [comment]
I can reproduce by generating a PDF directly through Chrome, so that means the bug is there. On further investigation, it looks like they are attempting to resolve rendering issues through https://bugs.chromium.org/p/chromium/issues/detail?id=829028

I'll keep an eye on this and report back when something changes.

**Comment by natewiebe13** [comment, marked as answer]
Between now and when this discussion started, the issue has been fixed on the Chrome side.
